crates/crabe_framework/src/data/geometry/penalty.rs

```diff
@@ -1,17 +1,43 @@
 use nalgebra::Point2;
 use serde::Serialize;
+use crabe_math::shape::Line;
 
-/// Represents a penalty area on a soccer field.
+/// Represents a penalty area on a soccer field. (all distances are in meters)
 #[derive(Serialize, Clone, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct Penalty {
-    /// The width of the penalty area in meters.
+    /// The width of the penalty area 
     pub width: f64,
-    /// The depth of the penalty area in meters.
+    /// The depth of the penalty area 
     pub depth: f64,
-    /// The top-left corner of the penalty area, measured from the origin of the
-    /// field, in meters.
-    pub top_left_position: Point2<f64>,
+    /// The front line of the penalty area
+    pub front_line: Line,
+    /// The left line of the penalty area (looking from the center of the field)
+    pub left_line: Line,
+    /// The right line of the penalty area (looking from the center of the field)
+    pub right_line: Line,
 }
 
-impl Penalty {}
+impl Penalty {
+
+    //return a penalty enlarged with the margin parameter 
+    pub fn enlarged_penalty(&self, offset: f64) -> Penalty {
```

Comment:
I think this function don't need to be on the penalty method.
It's an utility function for strategy not for the core part. 
I prefer to keep clean and minimalist the penalty and goal method.
Utility function for this can be added on the crabe_decision crate. 
For example for this method, we can do this : 
```rust
pub fn enlarged_penalty(&self, offset : f64, Penalty : &Penalty) -> Penalty {
```
What do you think about that ?

Reply:
I agree that it strange to have it inside the penalty data, 
However, if we put everything in the crabe_decision crate it will become a mess maybe, except if you have a subfolder called like "penalty_utils" or something like this